packages/multi-test/src/wasm.rs

```diff
@@ -719,25 +711,21 @@ where
         })
     }
 
-    pub fn load_contract(
-        &self,
-        storage: &dyn Storage,
-        address: &Addr,
-    ) -> Result<ContractData, String> {
+    pub fn load_contract(&self, storage: &dyn Storage, address: &Addr) -> Result<ContractData> {
         CONTRACTS
             .load(&prefixed_read(storage, NAMESPACE_WASM), address)
-            .map_err(|e| e.to_string())
+            .map_err(Into::into)
```

Comment:
And this will go away when we migrate everything to `anyhow`. :-)

Reply:
Do you want to change the contracts themselves???

They return eg. ContractError, which is nice and clear type.
We turn that into a string when we come to the wasm boundary.
I see turning ContractError to anyhow::Error very nice in the frame of multitest but this is the only place that we have multiple contracts communicating in rust and we need to handle many error types.

Reply:
I would not return anyhow from contracts directly. The idea about using it in place of strings. The most important downside is contract re-usability. We may want to execute functions from contracts in another contracts (it already happens), and it is good to have possibility to handling those error by their types. You can make an argument, that you can always have internal error type, and just return it as anyhow (and possibly downcast it when needed), but it is a regression. First problem is, that downcast has runtime overhead, which is 100% obsolete. The second, probably most important, that it is boilerplate - you need to do `downcast().unwrap()`, and you have ugly inwrap there which you should handle better (but why to handle it better when you are sure about internals?).

If `.map_err(Into::into)` is pain, then there is always a `Ok(res?)` which I personally find nasty (kind of strange way to express that what you want to do is just to map an error to more generic type). TBH I lack of function like `Result::map_into` which would be shorthand for doing what `?` does.

Reply:
OK, I understand. I was thinking of anyhow more like an equivalent / replacement for thiserror. Now I see that it is more like a complement.

Also, I thought that for the wasm interface, we can always convert anyhow to string, so that it would be equivalent to the current solution.